cmake: enable Fortran

opensuse tumbleweed:

`cmake -DBLAS_LIBRARIES=openblas ../lapack-3.12.0`

```
-- Looking for Fortran dgemm
CMake Error at /usr/share/cmake/Modules/CheckFortranFunctionExists.cmake:69 (try_compile):
  Unknown extension ".f" for file

    /tmp/build/CMakeFiles/CMakeScratch/TryCompile-6Nl0AU/testFortranCompiler.f

  try_compile() works only for enabled languages.  Currently these are:

    C CXX

  See project() command to enable other languages.
Call Stack (most recent call first):
  CMakeLists.txt:228 (CHECK_FORTRAN_FUNCTION_EXISTS)
```

cmake version is 3.27.7

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -223,6 +223,7 @@ option(USE_OPTIMIZED_BLAS "Whether or not to use an optimized BLAS library inste
 
 # Check the usage of the user provided BLAS libraries
 if(BLAS_LIBRARIES)
+  enable_language(Fortran)
   include(CheckFortranFunctionExists)
   set(CMAKE_REQUIRED_LIBRARIES ${BLAS_LIBRARIES})
   CHECK_FORTRAN_FUNCTION_EXISTS("dgemm" BLAS_FOUND)

